client/my-sites/plan-features/index.jsx

```diff
@@ -350,12 +351,19 @@ export class PlanFeatures extends Component {
 				relatedMonthlyPlan,
 				isPlaceholder,
 				hideMonthly,
+				rawPrice,
 			} = properties;
-			const { rawPrice, discountPrice } = properties;
+			let { discountPrice } = properties;
 			const classes = classNames( 'plan-features__table-item', 'has-border-top' );
 			let audience = planConstantObj.getAudience();
 			let billingTimeFrame = planConstantObj.getBillingTimeFrame();
 
+			if ( disableBloggerPlanWithNonBlogDomain || this.props.nonDotBlogDomains.length > 0 ) {
+				if ( planMatches( planName, { type: TYPE_BLOGGER } ) ) {
+					discountPrice = 0;
```

Comment:
Have you considered adjusting that on the API side?

Reply:
I looked into it, but decided against it as the plans endpoint is "dumb" in that it just returns a static list.  The shopping cart already does the validation part so it seemed like handling it on the frontend made the most sense.